Sources/Paths.swift

```diff
@@ -38,21 +38,22 @@ public struct PathPoint: Hashable {
 }
 
 extension PathPoint: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case position = "p"
+        case isCurved = "c"
+    }
+    
     public init(from decoder: Decoder) throws {
-        var container = try decoder.unkeyedContainer()
-        let x = try container.decode(Double.self)
-        let y = try container.decode(Double.self)
-        let z = (try? container.decodeIfPresent(Double.self) ?? 0) ?? 0
-        let isCurved = try container.decodeIfPresent(Bool.self) ?? false
-        self.init(Vector(x, y, z), isCurved: isCurved)
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let position = try container.decode(Vector.self, forKey: .position)
+        let isCurved = try container.decode(Bool.self, forKey: .isCurved)
+        self.init(position, isCurved: isCurved)
     }
 
     public func encode(to encoder: Encoder) throws {
-        var container = encoder.unkeyedContainer()
-        try container.encode(position.x)
-        try container.encode(position.y)
-        try position.z == 0 ? () : container.encode(position.z)
-        try isCurved ? container.encode(true) : ()
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(position, forKey: .position)
+        try container.encode(isCurved, forKey: .isCurved)
```

Comment:
This again increases the size of the encoded format by replacing the unkeyed flat format:

[x, y, z, true]

with a nested object format:

{ "p": { "x": x, "y": y, "z": z }, "c": true }